Marlin/src/feature/touch/xpt2046.cpp

```diff
@@ -82,12 +82,14 @@ uint8_t XPT2046::read_buttons() {
                  y = uint16_t(((uint32_t(getInTouch(XPT2046_Y))) * tsoffsets[2]) >> 16) + tsoffsets[3];
   if (!isTouched()) return 0; // Fingers must still be on the TS for a valid read.
 
-  if (y < 185 || y > 224) return 0;
-
-       if (WITHIN(x,  21,  98)) encoderDiff = -(ENCODER_STEPS_PER_MENU_ITEM) * ENCODER_PULSES_PER_STEP;
-  else if (WITHIN(x, 121, 198)) encoderDiff =   ENCODER_STEPS_PER_MENU_ITEM  * ENCODER_PULSES_PER_STEP;
-  else if (WITHIN(x, 221, 298)) return EN_C;
+  if (y < 175 || y > 234) return 0;
 
+       if (WITHIN(x,  11,  109)) encoderDiff = -(ENCODER_STEPS_PER_MENU_ITEM) * ENCODER_PULSES_PER_STEP;
+  else if (WITHIN(x, 111, 209)) encoderDiff =   ENCODER_STEPS_PER_MENU_ITEM  * ENCODER_PULSES_PER_STEP;
+  else if (WITHIN(x, 211, 309)) return EN_C;
+      #ifdef REVERSE_ENCODER_DIRECTION
+          encoderDiff = -encoderDiff;
+      #endif
```

Comment:
if you do that, need to edit dogm buttons too

Marlin/src/lcd/dogm/u8g_dev_tft_320x240_upscale_from_128x64.cpp
```
        #if DISABLED(REVERSE_ENCODER_DIRECTION)
          u8g_WriteEscSeqP(u8g, dev, button0_sequence);
          drawImage(button0, u8g, dev, 40, 20, TFT_BTSLEFT_COLOR);

          u8g_WriteEscSeqP(u8g, dev, button1_sequence);
          drawImage(button1, u8g, dev, 40, 20, TFT_BTSLEFT_COLOR);
        #else
          u8g_WriteEscSeqP(u8g, dev, button1_sequence);
          drawImage(button0, u8g, dev, 40, 20, TFT_BTSLEFT_COLOR);

          u8g_WriteEscSeqP(u8g, dev, button0_sequence);
          drawImage(button1, u8g, dev, 40, 20, TFT_BTSLEFT_COLOR);
        #endif
```

Reply:
Presumably (?) if the encoder is being reversed by a setting then it is to make the behavior correctly match the graphic. But whatever the case, let's be sure they match up.

Reply:
imo removing the ifdef is the easiest solution... make no sense else

Reply:
Should we remove just the `#ifdef` clause…?
```cpp
encoderDiff = -encoderDiff;
```
…or the whole block negating `encoderDiff`?